frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,188 @@
+  cy.url().should(
+    'include',
+    'k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins',
+  );
+  cy.get('.co-resource-item__resource-name')
+    .byLegacyTestID(PLUGIN_NAME)
+    .should('be.visible');
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Disabled' : 'Enabled');
+      cy.byTestID('edit-console-plugin').click();
+    });
+  modal.shouldBeOpened();
+  cy.contains('Cancel');
+  modal.modalTitleShouldContain('Console plugin enablement');
+  cy.byTestID(enable ? 'Enable-radio-input' : 'Disable-radio-input').click();
+  modal.submit();
+  modal.shouldBeClosed();
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Enabled' : 'Disabled');
+    });
+  cy.window().then((win: any) => {
+    if (win.SERVER_FLAGS?.authDisabled) {
+      // for local dev env just trigger any change in the return value to activate the toast
+      cy.intercept('/api/check-updates*', { plugins: [] }).as('checkUpdates');
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(CHECK_UPDATE_WAIT);
+      cy.wait('@checkUpdates');
+      cy.log('Running plugin test locally using bridge.');
+    } else {
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(CHECK_UPDATE_WAIT);
+      cy.log(`Running plugin test on ci using PLUGIN_PULL_SPEC: ${PLUGIN_PULL_SPEC}.`);
+    }
+    cy.get(warningToast)
+      .should('exist')
+      .then(() => {
+        cy.reload();
+        if (!enable) {
+          cy.byLegacyTestID(PLUGIN_NAME).click();
+          detailsPage.titleShouldContain(PLUGIN_NAME);
+          detailsPage.clickPageActionFromDropdown('Delete ConsolePlugin');
+          modal.shouldBeOpened();
+          modal.submit();
+        }
+      });
+  });
+};
+
+const dynamicNavTest = (navID: string) => {
+  nav.sidenav.clickNavLink(['Demo Plugin', `Dynamic Nav ${navID}`]);
+  cy.get('.pf-c-title.pf-m-2xl').should('contain', `Dynamic Page ${navID}`);
+  cy.get(infoMessage).should('exist');
+  cy.get(warningMessage).should('exist');
+  cy.get(hintMessage).should('exist');
+  cy.get('.pf-c-card').should('have.length', 50);
+};
+
+describe('Demo dynamic plugin test', () => {
+  before(() => {
+    cy.login();
+    nav.sidenav.switcher.changePerspectiveTo('Administrator');
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.visit('/k8s/all-namespaces/import');
+    cy.window().then((win: any) => {
+      const isLocalDevEnvironment = win.SERVER_FLAGS?.authDisabled;
+      cy.readFile(`${PLUGIN_PATH}/oc-manifest.yaml`).then((textManifest) => {
+        const yamlManifest = safeLoadAll(textManifest);
+        const yamlManifestUpdatedPullSpec = yamlManifest.map((yamlDefinition) => {
+          if (yamlDefinition.kind === 'Deployment' && !isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+            return _.defaultsDeep(
```

Comment:
You shouldn't need `_.defaultsDeep`. Instead of `map`, I would just find the deployment and set the value.

```ts
const deployment = yamlManifest.find(({kind}) => kind === 'Deployment');
deployment.spec.template.spec.containers[0].image = PLUGIN_PULL_SPEC;
```

Reply:
@spadgett thanks, using this to change the pull spec instead.